src/os/FileJournal.cc

```diff
@@ -1482,18 +1482,18 @@ void FileJournal::write_finish_thread_entry()
 {
 #ifdef HAVE_LIBAIO
   dout(10) << "write_finish_thread_entry enter" << dendl;
+  aio_lock.Lock();
   while (true) {
-    {
-      Mutex::Locker locker(aio_lock);
-      if (aio_queue.empty()) {
-	if (aio_stop)
-	  break;
-	dout(20) << "write_finish_thread_entry sleeping" << dendl;
-	write_finish_cond.Wait(aio_lock);
-	continue;
-      }
+    if (aio_queue.empty()) {
```

Comment:
old version:
unlock after write_finish_cond.Wait(aio_lock);
while(true)
lock before if (aio_queue.empty())

unlock/lock is unnecessary also